Enable validation of required case list `_all`

fedde-s opened the pull request:
A bug led to the validator sometimes failing to emit an error if this
case list is missing in a study, resulting in a server-side exception
(HTTP Status 500) when loading the OncoPrint. See issue #1214.
This commit fixes the bug and adds a unit test to prevent further
regressions.

In addition, fix the exception that occurs if a referenced data file cannot be found, and add a licence notice to the top of the file.

### Reviewers ###
@cBioPortal/core-developers

fedde-s commented:
@pieterlukasse and @zheins, could you review this?

pieterlukasse commented on core/src/main/scripts/importer/validateData.py:
```diff
@@ -2437,21 +2437,30 @@ def processCaseListDirectory(caseListDir, cancerStudyId, logger,
+        logger: logging.Logger instance to log output to
+    """
+
+    if cancer_study_id + '_all' not in case_list_fns:
```
shouldn't this if include a check on `add_global_case_list` as well? I.e. something like "if cancer_study_id + '_all' not in  case_list_fns **and** (`add_global_case_list` not set *or* set to false), then give error.

fedde-s replied:
No, in that case this dictionary will list that the case list with suffix `_all` was defined in the study metadata file.

fedde-s commented:
@pieterlukasse: I clarified the parts of the code that confused you, added a licence notice to the top of the file, and fixed the unrelated exception that we discussed.

fedde-s commented:
Travis is currently still failing because the test that was fixed in rc by PR #1657 was never fixed in hotfix.

jjgao commented:
Since there is no website code, I'll just merge it when it's ready. No worries about the screenshot tests. Please label it as "ready to merge" when ready.

pieterlukasse commented on core/src/main/scripts/importer/validateData.py:
```diff
@@ -2771,11 +2771,12 @@ def validate_study(study_dir, portal_instance, logger):
         logger.info("No directory named 'case_lists' found, so assuming no custom case lists.")
     else:
+        # add case lists IDs defined in the directory to any previous ones
         defined_case_list_fns = processCaseListDirectory(
```
maybe call this one defined_case_list_ids to keep code consistent?

fedde-s replied:
No, this is a dictionary that maps IDs to filenames.

pieterlukasse commented:
@fedde-s :+1: only one small comment.

pieterlukasse commented:
@jjgao ready to merge :+1:

fedde-s commented:
Now Travis is failing because of the frontend screenshot issue in pull request #1664.

fedde-s commented:
The Travis screenshot issues have been fixed upstream.
